runtime/v1/linux/task.go

```diff
@@ -99,7 +100,9 @@ func (t *Task) Delete(ctx context.Context) (*runtime.Exit, error) {
 		}
 	}
 	t.tasks.Delete(ctx, t.id)
-	if err := t.shim.KillShim(ctx); err != nil {
+	deferCtx, deferCancel := context.WithTimeout(context.TODO(), 1 * time.Minute)
```

Comment:
I think it is up to caller to pass `correct` context to handle the delete :)

Reply:
Why is this different from #4048 ?

Reply:
4048 is to cleanup in the defer and your change is not~. we can't change origin context in the main process.